src/httpserver.cpp

```diff
@@ -648,7 +652,11 @@ HTTPRequest::RequestMethod HTTPRequest::GetRequestMethod()
 void RegisterHTTPHandler(const std::string &prefix, bool exactMatch, const HTTPRequestHandler &handler)
 {
     LogPrint(BCLog::HTTP, "Registering HTTP handler for %s (exactmatch %d)\n", prefix, exactMatch);
-    pathHandlers.push_back(HTTPPathHandler(prefix, exactMatch, handler));
+    HTTPPathHandler pathHandler(prefix, exactMatch, handler);
+    if (std::find(pathHandlers.begin(), pathHandlers.end(), pathHandler) == pathHandlers.end()) {
```

Comment:
Thread https://github.com/bitcoin/bitcoin/pull/10650#discussion_r127313668

> I think that is something we can change later. Or do you think its crucial?

Not crucial (this isn't a bug and I already acked), but this seems very easy, unless I'm missing something?

```
auto it = find_if();
if (it == end) push_back(handler); else assert (it->handler == handler.handler);
```